blog/2022-08-29-web-app-demo/index.md

```diff
@@ -0,0 +1,265 @@
+resource "multy_database" "db" {
+  cloud          = var.cloud
+  location       = var.location
+  
+  storage_gb     = 10
+  name           = "multydb"
+  engine         = "mysql"
+  engine_version = "5.7"
+  username       = "multyadmin"
+  password       = random_password.password.result
+  size           = "micro"
+  subnet_id      = multy_subnet.subnet.id
+
+
+  depends_on = [multy_route_table_association.rta]
+}
+```
+
+
+## Configuring a virtual machine
+
+Finally, we need to configure the server. For this demo, we'll use the smallest server we can use. To initialize it, we'll make use of user data, a script that is executed by `cloud-init` when the virtual machine first boots.
+
+```hcl
+resource "multy_virtual_machine" "vm" {
+  cloud          = var.cloud
+  location       = var.location
+  
+  name               = "web_app_vm"
+  size               = "general_micro"
+  image_reference    = {
+    os      = "ubuntu"
+    version = "18.04"
+  }
+  subnet_id          = multy_subnet.subnet.id
+  generate_public_ip = true
+  user_data_base64   = base64encode(local.init_script)
+  # uncomment this line to allow ssh access
+  # public_ssh_key = file("~/.ssh/id_rsa.pub")
+  
+  network_security_group_ids = [multy_network_security_group.nsg.id]
+}
+
+locals {
+  init_script = <<EOT
+#!/bin/bash -xe
+sudo apt-get update -y && sudo apt-get -y install git npm mysql-client curl jq
+curl -fsSL https://deb.nodesource.com/setup_16.x | sudo -E bash -
+sudo apt-get install -y nodejs
+sudo chmod a+rwx .
+
+# putting secrets into user data is not best practice, you can use multy_vault instead
+export DATABASE_HOST=${multy_database.db.hostname}
+export DATABASE_USER=${multy_database.db.connection_username}
+export DATABASE_PASSWORD='${multy_database.db.password}'
+
+git clone https://github.com/FaztTech/nodejs-mysql-links.git
+cd nodejs-mysql-links
+
+mysql -h $DATABASE_HOST -P 3306 -u $DATABASE_USER --password=$DATABASE_PASSWORD -e 'source database/db.sql' || true
+
+npm i && npm run build && npm start
+EOT
+}
+
+output "endpoint" {
+  value = "http://${multy_virtual_machine.vm.public_ip}:4000"
+}
+```
+
+## Deploying
+
+After you have everything written, you can deploy the resources using terraform. Run the following command, using the cloud and [location](https://docs.multy.dev/regions) you prefer:
+
+```bash
+terraform apply -var="cloud=aws" -var="location=eu_west_1"
+```
+
+After a few minutes, you should see an endpoint where you can access the example app!
+
+## Summary
+
+Terraform is a very powerful tool that makes deployments repeatable using a single unified language (HCL) for any cloud. However, clouds are quite different from each other which means Terraform files look very different from cloud to cloud. In this tutorial we looked at an example on how to leverage Multy to make deployments portable across multiple cloud providers. If you want to explore other examples further, take a look at [Multy examples](https://docs.multy.dev/examples/). 
```

Comment:
When you say make deployments portable across multiple cloud providers.... Would it be worth it to say just changing your cloud parameter?

Reply:
Yea, good point